tc/sgx/trusted_worker_manager/enclave/work_order_data_handler.h

```diff
@@ -53,8 +53,11 @@ namespace tcf {
                 this->iv = iv;
             }
 
-            void Unpack(EnclaveData* enclaveData,
-                        const JSON_Object* object);
+            // enclave_type and data_encryption_key_from_pre_processing params
+            // are optional, used only by WPE
+            void Unpack(EnclaveData* enclaveData, const JSON_Object* object,
+                EnclaveType enclave_type=SINGLETON_ENCLAVE,
```

Comment:
Is this function is specific to WPE then can we use default enclave_type as WPE_ENCLAVE

Reply:
Also it might be worthwhile considering to use a sub class for wpe from WorkOrderDataHandler class and override `Unpack()` function.

Reply:
This function is common for all workers. Since the flow would deviate for WPE, i made default type to SINGLETON. I would consider creating separate WorkOrderDataHandler in a separate PR.